paddle/fluid/operators/elementwise/mkldnn/elementwise_add_mkldnn_op.cc

```diff
@@ -125,46 +131,44 @@ class EltwiseAddMKLDNNKernel : public framework::OpKernel<T> {
       std::vector<int> dst_tz = framework::vectorize2int(z_dims);
 
       std::vector<memory::primitive_desc> srcs_pd;
-      std::vector<memory> srcs;
+      std::vector<std::shared_ptr<mkldnn::memory>> srcs;
       std::vector<float> scales = {1.0f, 1.0f};
 
-      auto src_x_pd = memory::primitive_desc(
-          {{src_x_tz}, memory::data_type::f32, x->format()}, mkldnn_engine);
-      auto src_y_pd = memory::primitive_desc(
-          {{src_y_tz}, memory::data_type::f32, y->format()}, mkldnn_engine);
-      auto src_x_memory =
-          memory(src_x_pd, paddle::platform::to_void_cast(x_data));
-      auto src_y_memory =
-          memory(src_y_pd, paddle::platform::to_void_cast(y_data));
-
-      srcs_pd.push_back(src_x_pd);
-      srcs_pd.push_back(src_y_pd);
+      const std::string key = platform::MKLDNNHandler::GetHash(
+          src_x_tz, ctx.op().Output("Out") + std::to_string(x->format()) +
+                        std::to_string(y->format()));
+
+      platform::SumMKLDNNHandler handler(dev_ctx, mkldnn_engine, key);
+
+      auto src_x_memory = handler.AcquireSrcMemory(
+          {{src_x_tz}, platform::MKLDNNGetDataType<T>(), x->format()},
+          paddle::platform::to_void_cast(x_data));
+
+      auto src_y_memory = handler.AcquireSrc2Memory(
+          {{src_y_tz}, platform::MKLDNNGetDataType<T>(), y->format()},
+          paddle::platform::to_void_cast(y_data));
+
       srcs.push_back(src_x_memory);
       srcs.push_back(src_y_memory);
 
-      auto dst_md =
-          memory::desc({dst_tz}, memory::data_type::f32, memory::format::any);
+      auto dst_md = memory::desc({dst_tz}, platform::MKLDNNGetDataType<T>(),
+                                 memory::format::any);
 
-      // create primitive descriptor for sum
-      auto sum_pd = sum::primitive_desc(dst_md, scales, srcs_pd);
+      auto sum_pd = handler.AcquireSumPrimitiveDescriptor(srcs, scales, dst_md);
```

Comment:
why not use std::shared_ptr<>? this mkldnn memory will be stored in shared_ptr in blob, but it is not well cached or cache is cleaned, then this should be a potential core dump point.